src/Microsoft.ApplicationInsights/Extensibility/W3C/W3CActivityExtensions.cs

```diff
@@ -48,9 +37,10 @@ public static Activity GenerateW3CContext(this Activity activity)
         /// <param name="activity">Activity to check.</param>
         /// <returns>True if Activity has W3C properties, false otherwise.</returns>
         [EditorBrowsable(EditorBrowsableState.Never)]
+        [Obsolete("Activity from System.Diagnostics.DiagnosticSource 4.6.0 onwards natively support W3C.")]
         public static bool IsW3CActivity(this Activity activity)
-        {
-            return activity != null && activity.Tags.Any(t => t.Key == W3CConstants.TraceIdTag);
+        {            
+            return activity != null && Activity.DefaultIdFormat.Equals(ActivityIdFormat.W3C);
```

Comment:
nit: check the format on the Activity instance rather than static. not really important but may help one day